src/app/lightning/get-lightning-fee.ts

```diff
@@ -0,0 +1,80 @@
+import { checkedToSats, toSats } from "@domain/bitcoin"
+import {
+  decodeInvoice,
+  LnFeeCalculator,
+  RouteNotFoundError,
+} from "@domain/bitcoin/lightning"
+import { LnPaymentRequestZeroAmountRequiredError } from "@domain/errors"
+import { CachedRouteLookupKeyFactory } from "@domain/routes/key-factory"
+import { LndService } from "@services/lnd"
+import { RoutesCache } from "@services/redis/routes"
+
+export const lnInvoiceFeeProbe = async ({
+  paymentRequest,
+}: {
+  paymentRequest: EncodedPaymentRequest
+}): Promise<Satoshis | ApplicationError> => {
+  const decodedInvoice = await decodeInvoice(paymentRequest)
+  if (decodedInvoice instanceof Error) return decodedInvoice
+  const paymentAmount = checkedToSats(decodedInvoice.amount || 0)
+  if (paymentAmount instanceof Error) return paymentAmount
+
+  return feeProbe({ decodedInvoice, paymentAmount })
+}
+
+export const lnNoAmountInvoiceFeeProbe = async ({
+  paymentRequest,
+  amount,
+}: {
+  paymentRequest: EncodedPaymentRequest
+  amount: Satoshis
+}): Promise<Satoshis | ApplicationError> => {
+  const decodedInvoice = await decodeInvoice(paymentRequest)
+  if (decodedInvoice instanceof Error) return decodedInvoice
+
+  const { amount: lnInvoiceAmount } = decodedInvoice
+  if (lnInvoiceAmount && lnInvoiceAmount > 0) {
+    return new LnPaymentRequestZeroAmountRequiredError()
+  }
+
+  const paymentAmount = checkedToSats(amount)
+  if (paymentAmount instanceof Error) return paymentAmount
+
+  return feeProbe({ decodedInvoice, paymentAmount })
+}
+
+const feeProbe = async ({
+  decodedInvoice,
+  paymentAmount,
+}: {
+  decodedInvoice: LnInvoice
+  paymentAmount: Satoshis
+}): Promise<Satoshis | ApplicationError> => {
+  const { destination, paymentHash, milliSatsAmount } = decodedInvoice
+
+  const lndService = LndService()
+  if (lndService instanceof Error) throw lndService
+  if (lndService.isLocal(destination)) {
+    return toSats(0)
+  }
+
+  const key = CachedRouteLookupKeyFactory().create({
+    paymentHash,
+    milliSats: milliSatsAmount,
```

Comment:
what happen if is no amount invoice? we need to use paymentAmount

Reply:
Fixed https://github.com/GaloyMoney/galoy/pull/766/commits/1f3051187030ee5597d048cf5ea12c9b1c7100a4